types/subcall.go

```diff
@@ -1,15 +1,56 @@
 package types
 
-import "encoding/json"
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
 
-type replyOn string
+type replyOn int
 
 const (
-	ReplyAlways  replyOn = "always"
-	ReplySuccess replyOn = "success"
-	ReplyError   replyOn = "error"
+	ReplyAlways replyOn = iota
+	ReplySuccess
+	ReplyError
+	ReplyNever
 )
 
+var fromReplyOn = map[replyOn]string{
+	ReplyAlways:  "always",
+	ReplySuccess: "success",
+	ReplyError:   "error",
+	ReplyNever:   "never",
+}
+
+var toReplyOn = map[string]replyOn{
+	"always":  ReplyAlways,
+	"success": ReplySuccess,
+	"error":   ReplyError,
+	"never":   ReplyNever,
+}
+
+func (s replyOn) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(fromReplyOn[s])
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
```

Comment:
Can't use use something like `fromReplyOn[s].MarshalJSON()`?

Reply:
It doesn't look like that method is defined for a `string`. Huh.

Reply:
Yeah, turns out it is written differently: https://play.golang.org/p/QFEpCJ5j2o0